src/components/organisms/test/hooks/use-test.ts

```diff
@@ -16,14 +20,15 @@ import type { TestQuestionListType } from '../types';
  * */
 
 export const useTest = () => {
-  const { lessonId } = useParams();
+  const { lessonId }: { lessonId?: number } = useParams();
```

Comment:
Нет, параметры это всегда строки, и по умолчанию он типизирует верно.

```suggestion
  const { lessonId } = useParams();
```